app/src/main/java/com/jalasoft/routesapp/ui/home/view/HomeFragment.kt

```diff
@@ -125,27 +99,22 @@ class HomeFragment : HomeBaseFragment(), PossibleRouteAdapter.IPossibleRouteList
 
         // Next button
         binding.btnCheckNextLocation.setOnClickListener {
-            var newLocation: LatLng? = null
-            if (selectPointsStatus != SelectPointsStatus.BOTH) {
-                newLocation = getSelectedLocation()
-            }
             when (selectPointsStatus) {
                 SelectPointsStatus.ORIGIN -> {
+                    val newLocation = getSelectedLocation() ?: LatLng(0.0, 0.0)
+                    markerOrigin = addMarker(mMap!!, newLocation, R.drawable.ic_origin)
                     viewModel.setOrigin(newLocation)
-                    if (viewModel.selectedDestination.value == null) {
-                        selectPointsStatus = SelectPointsStatus.DESTINATION
-                    } else {
-                        isOriginLastSet = true
-                        selectPointsStatus = SelectPointsStatus.BOTH
-                    }
+                    selectPointsStatus = if (args.preSelectDestCoords != null) {
+                        markerDestination = addMarker(mMap!!, args.preSelectDestCoords!!, R.drawable.ic_destination)
```

Comment:
please avoid force unwrap, validate the variables with an elvis operator before the when statement